src/code/z_play.c

```diff
@@ -590,175 +613,184 @@ void Gameplay_Update(GlobalContext* globalCtx) {
                         } else {
                             globalCtx->transitionCtx.destroy(&globalCtx->transitionCtx);
                             func_800BC88C(globalCtx);
-                            globalCtx->transitionMode = 0;
+                            globalCtx->transitionMode = TRANS_MODE_OFF;
+
                             if (gTrnsnUnkState == 3) {
                                 TransitionUnk_Destroy(&sTrnsnUnk);
                                 gTrnsnUnkState = 0;
                                 R_UPDATE_RATE = 3;
                             }
                         }
-                        globalCtx->sceneLoadFlag = 0;
+
+                        globalCtx->transitionTrigger = TRANS_TRIGGER_OFF;
                     } else {
                         globalCtx->transitionCtx.update(&globalCtx->transitionCtx.data, R_UPDATE_RATE);
                     }
                     break;
             }
 
+            // update non-instance type transitions
```

Comment:
```suggestion
            // update non-instance mode transitions
```
feels weird to write but at this point the type bears no role, right?
can't think of better wording atm

Reply:
youre right that at this point the "mode" is what is referenced here, but its still pretty clearly seperated into two seperate block to handle instance types and non instance types. I dont think the switch needs to be on that variable specifically to say that the block of code is handling certain "types"

Reply:
for completeness, the switch also handles this single instance related mode `TRANS_MODE_INSTANCE_WAIT` so it's 99% non-instance, but not 100% lol

and sure if you understand the system it's clear the modes correspond to types at this point, but for discovering the system it's not immediately obvious the types are used to initialize the modes, so it may be confusing imo to assimilate types and modes

Reply:
imo wording it like this makes the separation sound more extreme than it actually is, handling "modes" just doesnt sound right to me personally. the types are what you interface with as the user and what youll be used to seeing them as

Reply:
then a compromise could be to not call them modes or types at all here but just
```suggestion
            // update non-instance transitions
```
lel